src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -0,0 +1,362 @@
+
+        findPaneOfTypeByID(BUTTON_PREVIOUS_STYLE_ID, Button.class).setEnabled(enabled);
+        findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class).setEnabled(enabled);
+        findPaneOfTypeByID(BUTTON_NEXT_STYLE_ID, Button.class).setEnabled(enabled);
+        stylesDropDownList.setSelectedIndex(newIndex);
+    }
+
+    /**
+     * Clears and resets/updates all resources.
+     */
+    private void updateResources()
+    {
+        final World world = Minecraft.getMinecraft().world;
+        resources.clear();
+
+        final StructureName sn = new StructureName(Structures.SCHEMATICS_PREFIX, styles.get(stylesDropDownList.getSelectedIndex()) , building.getSchematicName() + (building.getBuildingLevel() + 1));
+        final StructureWrapper wrapper = new StructureWrapper(world, sn.toString());
+        wrapper.setPosition(building.getLocation());
+        wrapper.rotate(building.getRotation(), world, building.getLocation(), building.isMirrored() ? Mirror.FRONT_BACK : Mirror.NONE);
+        while (wrapper.findNextBlock())
+        {
+            @Nullable final Template.BlockInfo blockInfo = wrapper.getBlockInfo();
+            @Nullable final Template.EntityInfo entityInfo = wrapper.getEntityinfo();
+
+            if (entityInfo != null)
+            {
+                for (final ItemStack stack : ItemStackUtils.getListOfStackForEntity(entityInfo, world, Minecraft.getMinecraft().player))
+                {
+                    if (!ItemStackUtils.isEmpty(stack))
+                    {
+                        addNeededResource(stack, 1);
+                    }
+                }
+            }
+
+            if (blockInfo == null)
+            {
+                continue;
+            }
+
+            @Nullable IBlockState blockState = blockInfo.blockState;
+            @Nullable Block block = blockState.getBlock();
+
+            if (wrapper.isStructureBlockEqualWorldBlock()
+                    || (blockState.getBlock() instanceof BlockBed && blockState.getValue(BlockBed.PART).equals(BlockBed.EnumPartType.FOOT))
+                    || (blockState.getBlock() instanceof BlockDoor && blockState.getValue(BlockDoor.HALF).equals(BlockDoor.EnumDoorHalf.UPPER)))
+            {
+                continue;
+            }
+
+            if (block != null
+                    && block != Blocks.AIR
+                    && !AbstractEntityAIStructure.isBlockFree(block, 0)
+                    && block != ModBlocks.blockSolidSubstitution
+                    && block != ModBlocks.blockSubstitution)
+            {
+                if (wrapper.getBlockInfo().tileentityData != null)
+                {
+                    final List<ItemStack> itemList = new ArrayList<>();
+                    if (wrapper.getBlockInfo() != null && wrapper.getBlockInfo().tileentityData != null)
+                    {
+                        itemList.addAll(ItemStackUtils.getItemStacksOfTileEntity(wrapper.getBlockInfo().tileentityData, world));
+                    }
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this line start at column 25. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)